snapcraft/plugins/v2/ruby.py

```diff
@@ -0,0 +1,156 @@
+      Additional configure options to use when configuring ruby.
+"""
+
+import os
+import re
+import logging
+
+from typing import Any, Dict, List, Set
+from snapcraft.plugins.v2 import PluginV2
+
+
+class RubyPlugin(PluginV2):
+    @classmethod
+    def get_schema(cls) -> Dict[str, Any]:
+        return {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "type": "object",
+            "additionalProperties": False,
+            "properties": {
+                "ruby-flavor": {
+                    "type": "string",
+                    "default": "ruby",
+                },
+                "ruby-version": {
+                    "type": "string",
+                    "default": "3.0",
+                    "pattern": r"^\d+\.\d+(\.\d+)?$",
+                },
+                "use-bundler": {
+                    "type": "boolean",
+                    "default": False,
+                },
+                "use-jemalloc": {
+                    "type": "boolean",
+                    "default": False,
+                },
+                "shared": {
+                    "type": "boolean",
+                    "default": False,
+                },
+                "configure-options": {
+                    "type": "array",
+                    "items": {"type": "string"},
+                    "default": [],
+                },
+                "gems": {
+                    "type": "array",
+                    "items": {"type": "string"},
+                    "default": [],
+                },
+            },
+        }
+
+    def get_build_snaps(self) -> Set[str]:
+        return set()
+
+    def get_build_packages(self) -> Set[str]:
+        packages = {"curl", "jq"}
+
+        if self.options.use_jemalloc:
+            packages.add("libjemalloc-dev")
+
+        return packages
+
+    def get_build_environment(self) -> Dict[str, str]:
+        env = {
+            "PATH": "${SNAPCRAFT_PART_INSTALL}/bin:${PATH}",
+        }
+
+        if self.options.shared:
+            # for finding ruby.so when running `gem` or `bundle`
+            env["LD_LIBRARY_PATH"] = "${SNAPCRAFT_PART_INSTALL}/lib:${LD_LIBRARY_PATH}"
+
+        return env
+
+    def _configure_opts(self) -> List[str]:
+        configure_opts = [
+            "--without-baseruby",
+            "--enable-load-relative",
+            "--disable-install-doc",
+        ] + self.options.configure_options
+
+        if self.options.shared:
+            configure_opts.append("--enable-shared")
+        if self.options.use_jemalloc:
+            configure_opts.append("--with-jemalloc")
+
+        return configure_opts
+
+    def _get_install_commands(self) -> List[str]:
+        commands = []
+        commands.append("ruby_install_url=$(curl -L --proto '=https' --tlsv1.2 'https://api.github.com/repos/postmodern/ruby-install/tags' | jq -r '.[0].tarball_url')")
+        commands.append("curl -L --proto '=https' --tlsv1.2 $ruby_install_url | tar xz")
```

Comment:
Is there anything we can do to authenticate this installer? signature perhaps?  Or maybe pin to a version via properties with an associated hash?

Reply:
That would be nice. I just checked the existing plugins. Looks like only V1 plugins use `sources.Tar` with the argument `source_checksum`. Can I do the same in this V2 plugin?

Reply:
@cjp256

Reply:
Solved with [984430c](https://github.com/snapcore/snapcraft/pull/3511/commits/984430c623f792434904e5f1635008a4829c9a5f)